packages/meditrak-server/src/routes/accessRequests/assertAccessRequestPermissions.js

```diff
@@ -19,6 +23,50 @@ export const assertAccessRequestPermissions = async (accessPolicy, models, acces
   return true;
 };
 
+export const assertAccessRequestEditPermissions = async (
+  accessPolicy,
+  models,
+  accessRequestId,
+  updatedFields,
+) => {
+  // Check we have basic permission to access the record we want to edit
+  await assertAccessRequestPermissions(accessPolicy, models, accessRequestId);
+  // Check we have permission for the change
+  await assertAccessRequestUpsertPermissions(accessPolicy, models, updatedFields);
+
+  // We can view access requests for BES admin access even if we don't have BES admin ourselves
+  // So this final check confirms we're not trying to approve a request for BES admin access
+  const accessRequest = await models.accessRequest.findById(accessRequestId);
+  const permissionGroup = await models.permissionGroup.findById(accessRequest.permission_group_id);
+  if (permissionGroup.name === BES_ADMIN_PERMISSION_GROUP) {
+    throw new Error('Need BES Admin access to make this change');
+  }
+
+  return true;
+};
+
+export const assertAccessRequestUpsertPermissions = async (
+  accessPolicy,
+  models,
+  { permission_group_id: permissionGroupId, entity_id: entityId },
+) => {
+  // Check we're not trying to change this access request to give someone:
+  // BES admin access
+  // Access to an entity we don't have admin panel access
+  if (permissionGroupId) {
+    const permissionGroup = await models.permissionGroup.findById(permissionGroupId);
+    if (permissionGroup.name === BES_ADMIN_PERMISSION_GROUP) {
+      throw new Error('Need BES Admin access to make this change');
+    }
+  }
+  if (entityId) {
+    const entity = await models.entity.findById(entityId);
+    if (!accessPolicy.allows(entity.country_code, TUPAIA_ADMIN_PANEL_PERMISSION_GROUP)) {
+      throw new Error('Need Admin Panel access to the updated entity');
```

Comment:
I'm seeing repeated code across this and one of the other PRs, but you know what - bugger it. I don't think it's worth making DRY.